File: apps/api/scripts/check-embeddings.ts
/** Read-only diagnosis. Model calls require the explicit --probe flag. */
import "dotenv/config";
import pg from "pg";
import { assertEmbeddingSchema } from "../src/lib/embedding-schema";
import { EMBEDDING_DIMENSIONS, embeddingDiagnostic, resolveEmbeddingConfig } from "../src/lib/embedding-contract";
import { embedQuery } from "../src/lib/embeddings";

const config = resolveEmbeddingConfig(process.env);
if (!process.env.DATABASE_URL) throw new Error("DATABASE_URL is required for embedding diagnosis.");
const client = new pg.Client({ connectionString: process.env.DATABASE_URL });
try {
  await client.connect();
  await assertEmbeddingSchema(client, config);
  if (process.argv.includes("--probe")) {
    const vector = await embedQuery(process.env, "VideoQ embedding preflight");
    console.info(JSON.stringify(embeddingDiagnostic(config, { db_dimensions: EMBEDDING_DIMENSIONS, actual_dimensions: vector.length })));
  }
} finally {
  await client.end();
}

File: apps/worker/worker_python/check_embeddings.py
"""Read-only diagnosis: python -m worker_python.check_embeddings [--probe]."""

import argparse
import json
import logging

from .pipeline.embedding_contract import EMBEDDING_DIMENSIONS, embedding_diagnostic, resolve_embedding_config
from .pipeline.embedding_schema import check_embedding_storage
from .pipeline.embeddings import embed_texts


def main() -> None:
    parser = argparse.ArgumentParser(description=__doc__)
    parser.add_argument("--probe", action="store_true", help="Call the configured model (may incur API charges).")
    args = parser.parse_args()
    logging.basicConfig(level=logging.INFO, format="%(message)s")
    config = resolve_embedding_config()
    check_embedding_storage()
    if args.probe:
        vector = embed_texts(["VideoQ embedding preflight"])[0]
        print(json.dumps(embedding_diagnostic(config, db_dimensions=EMBEDDING_DIMENSIONS, actual_dimensions=len(vector))))


if __name__ == "__main__":
    main()
